tests/test_cli_containerbuild.py

```diff
@@ -276,3 +377,156 @@ def test_compose_id_arg_restrictions(self, tmpdir, compose_ids, signing_intent,
 
         assert parsed_args == expected_args
         assert opts == expected_opts
+
+    @pytest.mark.parametrize('cause, stderr_msg', [
+        ('unknown target', 'Unknown build target'),
+        ('unknown dest tag', 'Unknown destination tag'),
+        ('dest tag locked', 'is locked'),
+        (':// not in URL', 'scm URL does not look like an URL to a source repository'),
+        ('# not in URL', 'scm URL must be of the form <url_to_repository>#<revision>)')
+    ])
+    def test_build_sysexit(self, cause, stderr_msg, capsys):
+        target = 'target'
+        source = 'https://repo#revision'
+        if cause == ':// not in URL':
+            source = source.replace('://', '')
+        elif cause == '# not in URL':
+            source = source.replace('#', '')
+
+        options = flexmock(quiet=True)
+
+        session = mock_session(
+            target,
+            source,
+            target_known=(cause != 'unknown target'),
+            dest_tag_known=(cause != 'unknown dest tag'),
+            dest_tag_locked=(cause == 'dest tag locked')
+        )
+
+        args = build_cli_args(target, source)
+
+        with pytest.raises(SystemExit):
+            cli_containerbuild.handle_container_build(options, session, args)
+
+        _, stderr_output = capsys.readouterr()
+        assert stderr_msg in stderr_output
+
+    @pytest.mark.parametrize('flatpak', [True, False])
+    @pytest.mark.parametrize('quiet', [True, False])
+    @pytest.mark.parametrize('build_result', [
+        {},
+        {'empty_list': []},
+        {'empty_dict': {}},
+        {'list': ['item_1', 'item_2'],
+         'dict': {'key': 'value'},
+         'key': 'value'},
+        {'list_of_dicts': [
+            {},
+            {'key': 'value'},
+            {'sublist': ['item']},
+            {'subdict': {'inner_key': 'inner_value'}}
+        ]},
+        {'list_of_lists': [
+            [],
+            ['item_1', 'item_2'],
+            [['nested']],
+            ['various_types', {'key': 'value'}]
+        ]}
+    ])
+    def test_build_success(self, build_result, quiet, flatpak, capsys):
+        target = 'target'
+        source = 'https://repo#revision'
+        task_id = '42'
+        # to prevent test failures due to dict iteration randomness
+        build_result = OrderedDict(build_result)
```

Comment:
Just realized the dicts should be turned into OrderedDicts recursively (doesn't matter for the current test cases, but just to be sure)